environment.py

```diff
@@ -65,14 +65,32 @@ def env() -> Mapping[str, Optional[str]]:
         # TDR source = 'tdr:', Google Cloud project name,
         #              ':', ( 'dataset' | 'snapshot' ),
         #              '/', TDR dataset or snapshot name,
-        #              ':', [ UUID prefix [ '/', Partition prefix length ] ] ;
+        #              ':', [ prefix ], [ '/', partition prefix length ] ;
         #
         # canned source = 'https://github.com',
         #                 '/', owner,
         #                 '/', repo,
         #                 '/tree/', ref,
-        #                 ['/', path] ;
-        #                 ':', [ UUID prefix [ '/', Partition prefix length ] ] ;
+        #                 ['/', path],
+        #                 ':', [ prefix ], [ '/', partition prefix length ] ;
+        #
+        # The `prefix` is an optional string of hexadecimal digits
+        # constraining the set of indexed subgraphs from the source. A
+        # subgraph will be indexed if its UUID begins with the `prefix`. The
+        # default `prefix` is the empty string.
+        #
+        # The partition prefix length is an optional integer that is used to
+        # further partition the set of indexed subgraphs. Each partition is
+        # assigned a prefix of `partition prefix length` hexadecimal digits.
+        # A subgraph belongs to a partition if its UUID starts with the
+        # overall `prefix` followed by the partition's prefix. The number of
+        # partitions of a source is therefore `16 ** partition prefix
+        # length`. The default `partition prefix length` is configured
+        # in the deployments environment.py file using the
+        # `AZUL_PARTITION_PREFIX_LENGTH` environment varaible.
```

Comment:
```suggestion
        # in the deployment's environment.py file using the
        # `AZUL_PARTITION_PREFIX_LENGTH` environment varaible.
```

But: This IS an `environment.py` file and the variable could also be defined here, not necessarily in the deployment-specific `environment.py`.

Reply:
```suggestion
        # in the deployments environment.py file using the
        # `AZUL_PARTITION_PREFIX_LENGTH` environment variable.
```

Reply:
> But: This IS an environment.py file and the variable could also be defined here, not necessarily in the deployment-specific environment.py.

We can refine this more but #3530 coming up behind this PR, this section will be refactored.

Reply:
> But: This IS an environment.py file and the variable could also be defined here, not necessarily in the deployment-specific environment.py.

I'd rather not get into the nuances of the env files here, in the actual main deployments we define it on a per-deployment basis, and this section is going to be refactored as soon as this merges in w/ changes from #3530.

On a side note it does not look like we have documentation about the precedence of the env files.